statsforecast/models.py

```diff
@@ -17,55 +19,141 @@
+                allowmean=False, allowdrift=False #not implemented yet
+            )
+        return self
+    
+    def predict(
+            self, 
+            h: int, # forecasting horizon 
+            X: np.ndarray = None, # exogenous regressors
+            level: Optional[Tuple[int]] = None, # level 
+        ):
+        fcst = forecast_arima(self.model_, h=h, xreg=X, level=level)
+        mean = fcst['mean']
+        if level is None:
+            return {'mean': mean}
+        return {
+            'mean': mean,
+            **{f'lo-{l}': fcst['lower'][f'{l}%'] for l in reversed(level)},
+            **{f'hi-{l}': fcst['upper'][f'{l}%'] for l in level},
+        }
+    
+    def predict_in_sample(self, level: Optional[Tuple[int]] = None):
+        if level is not None:
+            return NotImplementedError
+        mean = fitted_arima(self.model_)
         return {'mean': mean}
-    return {
-        'mean': mean,
-        **{f'lo-{l}': fcst['lower'][f'{l}%'] for l in reversed(level)},
-        **{f'hi-{l}': fcst['upper'][f'{l}%'] for l in level},
-    }
-
-# Cell
-def ets(X: np.ndarray, # time series
-        h: int, # forecasting horizon
-        future_xreg=None, # future regressors
-        fitted: bool = False, # fitted values
-        season_length: int = 1, # season length
-        model: str = 'ZZZ' # model type
-    )-> np.ndarray:
-    y = X[:, 0] if X.ndim == 2 else X
-    xreg = X[:, 1:] if (X.ndim == 2 and X.shape[1] > 1) else None
-    mod = ets_f(y, m=season_length, model=model)
-    fcst = forecast_ets(mod, h)
-    keys = ['mean']
-    if fitted:
-        keys.append('fitted')
-    return {key: fcst[key] for key in keys}
+    
+    def forecast(
+            self, 
+            y: np.ndarray, # time series
+            h: int, # forecasting horizon 
+            X: np.ndarray = None, # exogenous regressors
+            X_future: np.ndarray = None, # future regressors
+            level: Optional[Tuple[int]] = None, # level
+            fitted: bool = False, # return fitted values?
+        ):
+        with np.errstate(invalid='ignore'):
+            mod = auto_arima_f(
+                y,
+                xreg=X,
+                period=self.season_length,
+                approximation=self.approximation,
+                allowmean=False, allowdrift=False #not implemented yet
+            )
+        fcst = forecast_arima(mod, h, xreg=X_future, level=level)
+        res = {'mean': fcst['mean']}
+        if fitted:
+            res['fitted'] = fitted_arima(mod)
+        if level is not None:
+            res = {
+                **res,
+                **{f'lo-{l}': fcst['lower'][f'{l}%'] for l in reversed(level)},
+                **{f'hi-{l}': fcst['upper'][f'{l}%'] for l in level},
+            }
+        return res
+
+# %% ../nbs/models.ipynb 30
+class ETS(_TS):
+    
+    def __init__(self, season_length: int = 1, model: str = 'ZZZ'):
+        self.season_length = season_length
```

Comment:
Same comment that in `AutoARIMA`.
If users will be interacting mostly with the high-level class. It is convenient to add documentation to the constructor.

Is the ETS's fit method leveraging all the options from `ets_f`?